platform/zilliqa/rpc.go

```diff
@@ -15,37 +15,40 @@ func (c *RpcClient) GetBlockchainInfo() (info *ChainInfo, err error) {
 	return
 }
 
-func (c *RpcClient) GetTx(hash string) (tx Tx, err error) {
+func (c *RpcClient) GetTx(hash string) (tx TxRPC, err error) {
 	err = c.RpcCall(&tx, "GetTransaction", []string{hash})
 	return
 }
 
-func (c *RpcClient) GetTxInBlock(number int64) (txs []Tx, err error) {
-	strNumber := strconv.FormatInt(number, 10)
-	var results [][]string
-	err = c.RpcCall(&results, "GetTransactionsForTxBlock", []string{strNumber})
+func (c *RpcClient) GetTxInBlock(number int64) []Tx {
+	strNumber := strconv.Itoa(int(number))
+	txs := make([]Tx, 0)
+
+	var results BlockTxs
+	err := c.RpcCall(&results, "GetTransactionsForTxBlock", []string{strNumber})
+	if err != nil {
+		return txs
+	}
 
 	var wg sync.WaitGroup
-	out := make(chan Tx)
-	for _, ids := range results {
-		for _, id := range ids {
-			wg.Add(1)
-			go func(id string) {
-				defer wg.Done()
-				tx, errTx := c.GetTx(id)
-				if errTx != nil {
-					return
-				}
-				out <- tx
-			}(id)
-		}
+	btxs := results.txs()
+	out := make(chan Tx, len(btxs))
+	wg.Add(len(btxs))
+	for _, id := range btxs {
+		go func(hash string, txChan chan Tx, wg *sync.WaitGroup) {
+			defer wg.Done()
+			tx, errTx := c.GetTx(hash)
```

Comment:
I remember we did call batch JSON rpc? no need to use `sync.WaitGroup` at all

Reply:
zilliqa doesn't have a method to call the transactions in the batch

Reply:
This is JSON RPC, not view block api

Reply:
https://github.com/trustwallet/blockatlas/blob/084a6ee215efb6449e4d7a3b8ec53df417e64c25/platform/zilliqa/client.go#L68